### ISSUE TITLE
`imegeio.iv3.improps` fails on pretty standard tif file 

### ISSUE THREAD

**Issue by jmetz** [imageio/imageio#970]
Trying to get `improps` an a `tif` file that was created in `imagej`:

```pytb 
    props = iio.improps(filepath)
  File "/home/jeremy/.local/lib/python3.10/site-packages/imageio/v3.py", line 198, in improps
    properties = img_file.properties(**call_kwargs)
  File "/home/jeremy/.local/lib/python3.10/site-packages/imageio/plugins/tifffile_v3.py", line 379, in properties
    spacing=_get_resolution(target_page)["resolution"],
KeyError: 'resolution'
```

Presumably an easy fix would be to use `get(...)` an that dict access in `properties` instead of `[...]`, with an appropriate fallback?

Otherwise the fix would need to look at why `_get_resolution` is returning a dict without the `"resolution"` key :shrug:

**Comment by jmetz** [comment]
PS: I saved the same image using _"save as OME-tif"_ in Imagej and `improps` works on that!

**Comment by jmetz** [comment]
Looking at `_get_resolution` seems to imply some foreknowledge of this issue: 
https://github.com/imageio/imageio/blob/master/imageio/plugins/tifffile_v3.py#L62-L67

```python
    metadata = {
        # uncomment once py 3.7 reached EoL - in fact, refactor this
        # function :)
        # "resolution_unit": page.resolutionunit,
        # "resolution": page.resolution,
    }
```

As `3.7`'s EoL is only a couple months away (2023-06-27, https://devguide.python.org/versions/#supported-versions) I guess as a fallback to fixing this, this issue could be put on hold until then, as long as then the suggestion in the comments is implemented :shrug:  

@FirefoxMetzger  - you authored this, I don't fully understand why the fallback values were commented out with the comment about `py37` - any chance you could comment?

**Comment by FirefoxMetzger** [comment]
Of course @jmetz . I will have to delay taking a more in-depth look at this issue until I have more time on my hands, but I can comment on ImageIO's source.

The reason this section is commented out is because the version of tifffile needed to access `page.resolutionunit` and `page.resolution` requires python 3.8+. ImageIO supports all active Python versions, so we have to wait until 3.7 is phased out before we can safely rely on these properties.

We could - of course - check the current version of tifffile or the interpreter and base our logic on that, but since EoL for 3.7 is around the corner I thought that it is easier to wait unless someone has a pressing need for it.

**Comment by jmetz** [comment]
Thanks for the speedy reply @FirefoxMetzger - in the meantime I just monkey-patched `imageio.v3` to suit my needs within my own module:

```python 
# Monkeypatch imageio as of 2023-04-25 as _get_resolution fails
import imageio.plugins.tifffile_v3


def _get_resolution(page):
    """Get the resolution in a py3.7 compatible way"""

    metadata = {
        # uncomment once py 3.7 reached EoL - in fact, refactor this
        # function :)
        "resolution_unit": page.resolutionunit,
        "resolution": page.resolution,
    }

    if 296 in page.tags:
        metadata["resolution_unit"] = page.tags[296].value.value

    if 282 in page.tags and 283 in page.tags and 296 in page.tags:
        resolution_x = page.tags[282].value
        resolution_y = page.tags[283].value
        if resolution_x[1] == 0 or resolution_y[1] == 0:
            warnings.warn(
                "Ignoring resulution metadata, "
                "because at least one direction has a 0 denominator.",
                RuntimeWarning,
            )
        else:
            metadata["resolution"] = (
                resolution_x[0] / resolution_x[1],
                resolution_y[0] / resolution_y[1],
            )

    return metadata


imageio.plugins.tifffile_v3._get_resolution = _get_resolution

```

So far it seems to do what I need it to, ie not cause the `KeyError` in later calls to `imageio.v3.improp`, so would recommend this as an interim fix for anyone with the same issue.
